lib/internal/url.js

```diff
@@ -214,6 +217,14 @@ class URL {
     return (this[context].flags & binding.URL_FLAGS_CANNOT_BE_BASE) !== 0;
   }
 
+  // https://url.spec.whatwg.org/#cannot-have-a-username-password-port
+  get [cannotHaveUsernamePasswordPort]() {
+    const { host, scheme } = this[context];
+    return ((host == null || host === '') ||
```

Comment:
The spec says "if its host is null or the empty string" so it'll be clearer this way, and no it's not faster in my tests.